### PR TITLE
Fix import location hierarchy

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | yes/no
| New feature?  | yes/no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/glpi-project/glpi-inventory-plugin/issues/513 (and maybe #17106)

### CODE REVIEW

**Comment by trasher** [comment]
I'm a bit surprised location is created with the full name; I was ewpecting its name to be "Paris" only.

I'll dig into that if necessary when I'll be back.

**Comment by khaegebaert** [comment]
Hi, I've tried your solution as I have the issue with locations for users and inventory but got an error: 

[2024-06-03 14:05:16] glpiphplog.CRITICAL:   *** Uncaught Exception TypeError: str_replace(): Argument #3 ($subject) must be of type array|string, LDAP\Connection given in /opt/glpi/src/Html.php at line 318
  Backtrace :
  src/Html.php:318                                   str_replace()
  :                                                  Html::cleanPostForTextArea()
  src/Html.php:304                                   array_map()
  src/CommonDBTM.php:1187                            Html::cleanPostForTextArea()
  src/CommonDBTM.php:1736                            CommonDBTM->restoreInput()
  src/AuthLDAP.php:2970                              CommonDBTM->update()
  src/AuthLDAP.php:404                               AuthLDAP::ldapImportUserByServerId()
  src/MassiveAction.php:1420                         AuthLDAP::processMassiveActionsForOneItemtype()
  src/MassiveAction.php:1398                         MassiveAction->processForSeveralItemtypes()
  front/massiveaction.php:62                         MassiveAction->process()
  public/index.php:82                                require()

**Comment by trasher** [comment]
Either this is not related; either the proposed solution is not correct at all.

**Comment by trasher** [comment]
I cannot test LDAP import, but if proposed changes here cause an `LDAP\Connection` to be sent instead of a string; I guess there is a problem elsewhere.
Please also make sure you do not have plugins installed.

File: src/CommonTreeDropdown.php [hunk 1]
```diff
@@ -937,13 +937,13 @@ public function findID(array &$input)
 
     public function import(array $input)
     {
-
-        if (isset($input['name'])) {
-            return parent::import($input);
+        if (!isset($input['name']) && !isset($input['completename'])) {
```
Comment (cedric-anne) [inline src/CommonTreeDropdown.php:940]:
```suggestion
        if (empty($input['name']) && empty($input['completename'])) {
```

File: src/CommonTreeDropdown.php [hunk 2]
```diff
@@ -937,13 +937,13 @@ public function findID(array &$input)
...
+            return -1;
         }
 
         if (!isset($input['completename']) || empty($input['completename'])) {
```
Comment (cedric-anne) [inline src/CommonTreeDropdown.php:944]:
```suggestion
        if (empty($input['completename'])) {
```

**Comment by khaegebaert** [comment]
Hi, tried the changes from cedric-anne, everything ok now,  solves the bug and no errors. 
Thanks

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**